cuda/matrix/csr_kernels.cu

```diff
@@ -1447,6 +1447,49 @@ void extract_diagonal(std::shared_ptr<const CudaExecutor> exec,
 GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(GKO_DECLARE_CSR_EXTRACT_DIAGONAL);
 
 
+template <typename ValueType, typename IndexType>
+void check_diagonal_entries_exist(
+    std::shared_ptr<const CudaExecutor> exec,
+    const matrix::Csr<ValueType, IndexType>* const mtx,
+    bool* const has_all_diags)
+{
+    const size_type num_blocks = exec->get_num_multiprocessor() * 4;
+    Array<int> block_has_diags(exec, num_blocks);
+    kernel::check_diagonal_entries<<<num_blocks, default_block_size>>>(
```

Comment:
```suggestion
if (num_blocks > 0) 
    kernel::check_diagonal_entries<<<num_blocks, default_block_size>>>(
```
same for the other kernel launches with variabel number of blocks

Reply:
We never seem to have done this before. Was the assumption that in practice the number of blocks would not be zero because of the use of `ceildiv`? In this particular location, we would always have a nonzero number of SMs, so it should be fine.

Reply:
ceildiv(0,...) is still 0, and empty kernel launches cause CUDA to return an error code from cudaGetLastError. I added guards like these to all kernel launches in #905, and should probably establish this elsewhere as well.

Reply:
This still needs to be addressed